### PR TITLE
Refactored querying of default entity into configuration manger service.

### PR DESCRIPTION
**Pull request by markdorison**
- Created configuration manager service.
- Created constant to store default entity type.
- Updated calling code to use configuration manager service instead of querying configuration directly.

@see https://www.drupal.org/node/2649790

@ChromaticHQ/recurly And I thought this was going to be a small one..

### CODE REVIEW

**Comment by adamzimmermann** [comment]
@markdorison can we walk through this on the contrib standup tomorrow? There are a lot of changes here, some I understand, some I'm not fully grasping. They all look good and seem to exemplify a lot of D8 best practices that I'm trying to get my head around.

**Comment by markdorison** [comment]
Ready for another look. cc/ @adamzimmermann

File: modules/recurlyjs/src/Plugin/Derivative/RecurlyJsLocalTask.php [hunk 1]
```diff
@@ -47,7 +58,8 @@ public function __construct(EntityManagerInterface $entity_manager, TranslationI
   public static function create(ContainerInterface $container, $base_plugin_id) {
     return new static(
       $container->get('entity.manager'),
-      $container->get('string_translation')
+      $container->get('string_translation'),
+      $container->get('recurly.config_manager')
```
Comment (adamzimmermann) [inline modules/recurlyjs/src/Plugin/Derivative/RecurlyJsLocalTask.php:62]: I wish you could add commas after the last argument so they were consistent with multiline arrays.

File: recurly.services.yml [hunk 2]
```diff
@@ -3,42 +3,42 @@ services:
     class: Drupal\recurly\Access\RecurlyAccessLocalAccount
     tags:
       - { name: access_check, applies_to: _access_check_recurly_local_account }
-    arguments: ['@current_route_match']
+    arguments: ['@current_route_match', '@recurly.config_manager']
```
Comment (adamzimmermann) [inline recurly.services.yml:6]: Interesting/annoying that you have to declare this here and in the `create()` method.

File: src/Form/RecurlySubscriptionPlansForm.php [hunk 3]
```diff
@@ -7,52 +7,14 @@
...
 use Drupal\Component\Utility\SafeMarkup;
-use Symfony\Component\DependencyInjection\ContainerInterface;
-use Drupal\recurly\RecurlyFormatManager;
-use Drupal\recurly\RecurlyUrlManager;
+use Drupal\recurly\Form\RecurlyFormBase;
 
 /**
  * Recurly subscription plans form.
  */
-class RecurlySubscriptionPlansForm extends FormBase {
```
Comment (adamzimmermann) [inline src/Form/RecurlySubscriptionPlansForm.php:20]: ++ for putting shared form functionality together too.

**Comment by adamzimmermann** [comment]
:+1:

File: recurly.services.yml (hunk 2 above)
Comment (markdorison) [inline recurly.services.yml:6]: You don't. There is no `create()` method in that class.

File: modules/recurlyjs/src/Plugin/Derivative/RecurlyJsLocalTask.php (hunk 1 above)
Comment (markdorison) [inline modules/recurlyjs/src/Plugin/Derivative/RecurlyJsLocalTask.php:62]: ¯_(ツ)_/¯

It's a method call, after all.

Comment (adamzimmermann) [inline modules/recurlyjs/src/Plugin/Derivative/RecurlyJsLocalTask.php:62]: #firstworldproblems

**Comment by markdorison** [comment]
Patch posted: https://www.drupal.org/node/2649790#comment-10844184

File: recurly.services.yml (hunk 2 above)
Comment (adamzimmermann) [inline recurly.services.yml:6]: Refresh me on what this does for us then?

Comment (markdorison) [inline recurly.services.yml:6]: This creates it for us when passing it to another service. I believe we need to instantiate it ourselves when it is not a service we are passing to.

Comment (adamzimmermann) [inline recurly.services.yml:6]: Ah ok, so services are treated differently in regards to how their dependencies are injected. Got it!

:+1:

Comment (markdorison) [inline recurly.services.yml:6]: @adamzimmermann I believe they just have some extra Symfony sugar to sweeten them.

**Pull request closed without merging**